examples/webhook.php

```diff
@@ -8,67 +8,165 @@
+        $headers = getallheaders();
+        $sig = $headers['Btcpay-Sig'];
+
+        if ($sig !== "sha256=" . hash_hmac('sha256', $raw_post_data, $this->secret)) {
+            fwrite(
+                $myfile,
+                $date . " : Error. Invalid Signature detected! \n was: " . $sig . " should be: " . hash_hmac(
+                    'sha256',
+                    $raw_post_data,
+                    $this->secret
+                ) . "\n"
+            );
+            fclose($myfile);
+            throw new \Exception(
+                'Invalid BTCPayServer payment notification message received - signature did not match.'
+            );
+        }
+
+        if (true === empty($payload->invoiceId)) {
+            fwrite(
+                $myfile,
+                $date . " : Error. Invalid BTCPayServer payment notification message received - did not receive invoice ID.\n"
+            );
+            fclose($myfile);
+            throw new \Exception(
+                'Invalid BTCPayServer payment notification message received - did not receive invoice ID.'
+            );
+        }
+
+        // Load an existing invoice with the provided invoiceId.
+        // Most of the time this is not needed as you can listen to specific webhook events
+        // See: https://docs.btcpayserver.org/API/Greenfield/v1/#tag/Webhooks/paths/InvoiceCreated/post
+        try {
+            $client = new Invoice($this->host, $this->apiKey);
+            $invoice = $client->getInvoice($this->storeId, $payload->invoiceId);
+        } catch (\Throwable $e) {
+            fwrite($myfile, "Error: " . $e->getMessage());
+            throw $e;
+        }
+
+        // optional: check whether your webhook is of the desired type
+        if ($payload->type !== "InvoiceSettled") {
+            throw new \Exception(
+                'Invalid payload message type. Only InvoiceSettled is supported, check the configuration of the webhook.'
+            );
+        }
+
+        $invoicePrice = $invoice->getData()['amount'];
+        $buyerEmail = $invoice->getData()['metadata']['buyerEmail'];
+
+        fwrite(
+            $myfile,
+            $date . " : Payload received for BtcPay invoice " . $payload->invoiceId . " Type: " . $payload->type . " Price: " . $invoicePrice . " E-Mail: " . $buyerEmail . "\n"
+        );
+        fwrite($myfile, "Raw payload: " . $raw_post_data . "\n");
+
+        // your own processing code goes here!
+
+        echo 'OK';
+    }
+
+    public function createWebhook()
+    {
+        $url = 'https://createdurl.test.example.com/webhook';
+        $specificEvents = [
+          'InvoiceExpired',
+          'InvoiceSettled',
+          'InvoiceInvalid'
+        ];
+
+        try {
+            $client = new \BTCPayServer\Client\Webhook($this->host, $this->apiKey);
+            var_dump($client->createWebhook($this->storeId, $url, $specificEvents, null));
+        } catch (\Throwable $e) {
+            echo "Error: " . $e->getMessage();
```

Comment:
Other methods are throwing the exception but this one doesn't?

Reply:
this is how we do it on all other examples so I kept doing that the other code is just moved "as is"